File: src/test/java/io/femo/http/transport/http2/HttpFrameTest.java
package io.femo.http.transport.http2;

import io.femo.http.Constants;
import org.junit.Test;

import static org.junit.Assert.*;

/**
 * Created by felix on 9/3/16.
 */
public class HttpFrameTest {

    @Test(expected = HttpFrameException.class)
    public void testBiggerThanInSettings() throws Exception {
        HttpSettings settings = new HttpSettings(HttpSettings.EndpointType.SERVER);
        settings.setMaxFrameSize(20);
        HttpFrame frame = new HttpFrame(settings);
        frame.setLength(30);
    }

    @Test(expected = HttpFrameException.class)
    public void testBiggerThanInSettingsDefault() throws Exception {
        HttpSettings settings = new HttpSettings(HttpSettings.EndpointType.SERVER);
        HttpFrame frame = new HttpFrame(settings);
        frame.setLength(17000);
    }
    @Test(expected = HttpFrameException.class)
    public void testBiggerThanMaxPossible() throws Exception {
        HttpSettings settings = new HttpSettings(HttpSettings.EndpointType.SERVER);
        settings.setMaxFrameSize(Integer.MAX_VALUE);
        HttpFrame frame = new HttpFrame(settings);
        frame.setLength(Integer.MAX_VALUE);
    }

    @Test(expected = HttpFrameException.class)
    public void testTypeLimitations() throws Exception {
        HttpSettings settings = new HttpSettings(HttpSettings.EndpointType.SERVER);
        HttpFrame frame = new HttpFrame(settings);
        frame.setType((short) 300);
    }

    @Test(expected = HttpFrameException.class)
    public void testTypeNegative() throws Exception {
        HttpSettings settings = new HttpSettings(HttpSettings.EndpointType.SERVER);
        HttpFrame frame = new HttpFrame(settings);
        frame.setType((short) -12);
    }

    @Test(expected = HttpFrameException.class)
    public void testFlagsLimitations() throws Exception {
        HttpSettings settings = new HttpSettings(HttpSettings.EndpointType.SERVER);
        HttpFrame frame = new HttpFrame(settings);
        frame.setFlags((short) 300);
    }

    @Test(expected = HttpFrameException.class)
    public void testFlagsNegative() throws Exception {
        HttpSettings settings = new HttpSettings(HttpSettings.EndpointType.SERVER);
        HttpFrame frame = new HttpFrame(settings);
        frame.setFlags((short) -12);
    }

    @Test(expected = HttpFrameException.class)
    public void testStreamIdentifierNegative() throws Exception {
        HttpSettings settings = new HttpSettings(HttpSettings.EndpointType.SERVER);
        HttpFrame frame = new HttpFrame(settings);
        frame.setStreamIdentifier(-12);
    }
}

File: src/test/java/org/web25/http/transport/http2/HttpFrameTest.kt
package org.web25.http.transport.http2

import org.junit.jupiter.api.Assertions.assertThrows
import org.junit.jupiter.api.Test


/**
 * Created by felix on 9/3/16.
 */
internal class HttpFrameTest {

    @Test
    fun testBiggerThanInSettings() {
        val settings = HttpSettings(HttpSettings.EndpointType.SERVER)
        settings.maxFrameSize = 20
        val frame = HttpFrame(settings)
        assertThrows<HttpFrameException>(HttpFrameException::class.java, {
            frame.length = 30
        })
    }

    @Test
    fun testBiggerThanInSettingsDefault() {
        val settings = HttpSettings(HttpSettings.EndpointType.SERVER)
        val frame = HttpFrame(settings)
        assertThrows<HttpFrameException>(HttpFrameException::class.java, {
            frame.length = 17000
        })
    }

    @Test
    fun testBiggerThanMaxPossible() {
        val settings = HttpSettings(HttpSettings.EndpointType.SERVER)
        settings.maxFrameSize = Integer.MAX_VALUE
        val frame = HttpFrame(settings)
        assertThrows<HttpFrameException>(HttpFrameException::class.java, {
            frame.length = Integer.MAX_VALUE
        })
    }

    @Test
    fun testTypeLimitations() {
        val settings = HttpSettings(HttpSettings.EndpointType.SERVER)
        val frame = HttpFrame(settings)
        assertThrows<HttpFrameException>(HttpFrameException::class.java, {
            frame.type = 300.toShort()
        })
    }

    @Test
    fun testTypeNegative() {
        val settings = HttpSettings(HttpSettings.EndpointType.SERVER)
        val frame = HttpFrame(settings)
        assertThrows<HttpFrameException>(HttpFrameException::class.java, {
            frame.type = (-12).toShort()
        })
    }

    @Test
    fun testFlagsLimitations() {
        val settings = HttpSettings(HttpSettings.EndpointType.SERVER)
        val frame = HttpFrame(settings)
        assertThrows<HttpFrameException>(HttpFrameException::class.java, {
            frame.flags = 300.toShort()
        })
    }

    @Test
    fun testFlagsNegative() {
        val settings = HttpSettings(HttpSettings.EndpointType.SERVER)
        val frame = HttpFrame(settings)
        assertThrows<HttpFrameException>(HttpFrameException::class.java, {
            frame.flags = (-12).toShort()
        })
    }

    @Test
    fun testStreamIdentifierNegative() {
        val settings = HttpSettings(HttpSettings.EndpointType.SERVER)
        val frame = HttpFrame(settings)
        assertThrows<HttpFrameException>(HttpFrameException::class.java, {
            frame.streamIdentifier = -12
        })
    }
}